lib/ansible/plugins/httpapi/fortimanager.py

```diff
@@ -281,26 +291,32 @@ def check_mode(self):
         Checks FortiManager for the use of Workspace mode
         """
         url = "/cli/global/system/global"
-        code, resp_obj = self.send_request(FMGRMethods.GET, self._tools.format_request(FMGRMethods.GET, url,
-                                                                                       fields=["workspace-mode",
-                                                                                               "adom-status"]))
+        code, resp_obj = self.send_request(FMGRMethods.GET,
+                                           self._tools.format_request(FMGRMethods.GET,
+                                                                      url,
+                                                                      fields=["workspace-mode", "adom-status"]))
         try:
-            if resp_obj["workspace-mode"] != 0:
+            if resp_obj["workspace-mode"] == "workflow":
                 self.uses_workspace = True
+            elif resp_obj["workspace-mode"] == "disabled":
+                self.uses_workspace = False
         except KeyError:
-            self.uses_workspace = False
+            raise FMGBaseException(msg="Couldn't determine workspace-mode in the plugin")
         try:
-            if resp_obj["adom-status"] == 1:
+            if resp_obj["adom-status"] in [1, "enable"]:
                 self.uses_adoms = True
+            else:
+                self.uses_adoms = False
         except KeyError:
-            self.uses_adoms = False
+            raise FMGBaseException(msg="Couldn't determine adom-status in the plugin")
 
     def run_unlock(self):
         """
         Checks for ADOM status, if locked, it will unlock
         """
-        for adom_locked in self._locked_adom_list:
-            self.unlock_adom(adom_locked)
+        for adom_locked in self._locked_adoms_by_user:
+            adom = adom_locked["adom"]
+            self.unlock_adom(adom)
 
     def lock_adom(self, adom=None, *args, **kwargs):
```

Comment:
hi Luke, 
    I see from the code, you query the lock info and try to unlock the adoms upon logging out.
    however, I don't see any places where lock_adom() is called to lock a domain. 
    if we don't explicitly lock the domain in our plugin, is it required to unlock it every time when the plugin is logging out?

+ @frankshen01 

thanks,
Link

Reply:
It's a measure to make sure that it gets unlocked, always. If the session that does the locking, doesn't unlock it, it's hard to reverse that. So no matter what, Ansible has to unlock when its done. This is another layer to make sure that happens.